### ISSUE TITLE
[BUG] testhost is locking files during build

### ISSUE THREAD

**Issue by jaredpar** [microsoft/vscode-dotnettools#1089]
### Describe the Issue

When building my solution I frequently get the following warnings:

```txt
C:\Program Files\dotnet\sdk\8.0.104\Microsoft.Common.CurrentVersion.targets(4911,5): warning MSB3026: Could not copy "C:\Users\jaredpar\code\wt\ros1\artifac 
ts\bin\Roslyn.Test.PdbUtilities\Debug\net472\Roslyn.Test.PdbUtilities.dll" to "C:\Users\jaredpar\code\wt\ros1\artifacts\bin\Microsoft.CodeAnalysis.CSharp.Se 
mantic.UnitTests\Debug\net472\Roslyn.Test.PdbUtilities.dll". Beginning retry 5 in 1000ms. The process cannot access the file 'C:\Users\jaredpar\code\wt\ros1 
\artifacts\bin\Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests\Debug\net472\Roslyn.Test.PdbUtilities.dll' because it is being used by another process. The  
file is locked by: "testhost (24424)" [C:\Users\jaredpar\code\wt\ros1\src\Compilers\CSharp\Test\Semantic\Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests.cs 
proj::TargetFramework=net472]
```

This is frustrating because it's hiding other warnings. Furthermore this often leads to errors when MSBuild exceeds its retry count. 

Really hampers my ability to use VS Code to work with C#

### Steps To Reproduce

1. Clone https://github.com/dotnet/roslyn
2. Make an edit
3. Use the `build Compilers.slnf` build task

Repeating steps 2 and 3 will eventually cause the warning to be hit. 

Note: you likely need to install C# DevKit to hit this repro

### Expected Behavior

No warnings.

**Comment by arkalyanms** [comment]
Starting with area-project. @tmeschter would you please route?

**Comment by tmeschter** [comment]
Rerouting to "area-test" as the files in question are locked by testhost.

**Comment by ocallesp** [comment]
I was unable to repro this bug.

I opened the solution Compilers.slnf then made a code change in PdbTestUtilities.cs, which is part of PdbUtilities within Tests, then build. 

I tried like 10 times with Build and Rebuild but was unsuccessful.

**Comment by ocallesp** [comment]
Potential solutions:
- We can enhance this scenario by some events from msbuild to cancel the discovery on build.

- we can try to implement a mechanism to disable automatic discovery and trigger it as part of the refresh button.

It seems second option.

**Comment by jaredpar** [comment]
> I was unable to repro this bug.

What can I do to help with the repro? Is there extra logging I can turn on, binary logs, etc ... 

> We can enhance this scenario by some events from msbuild to cancel the discovery on build.

Curious: how are you opening the files here? Are you using `FileShare.Read`? Both processes are reading the binaries so naiively I would expect that concurrent reads shouldn't bea problem. 

> we can try to implement a mechanism to disable automatic discovery and trigger it as part of the refresh button.

That definitely could be an option.

**Comment by jaredpar** [comment]
Once again trying to use DevKit today and blocked by this issue. I have to resort to "just keep building until it eventually works". 

What can I provide to help track this down? It repros across every dev machine I have, it's a **constant** blocker for using DevKit.

**Comment by peterwald** [comment]
I've added a configuration option to C# Dev Kit that turns off auto-discovery after build. This makes things a bit more manual, but you shouldn't have file locking issues caused by discovery and build clashing.

To make this change set `dotnet.testWindow.disableAutoDiscovery=true`

When auto-discovery is disabled, tests will not be discovered after building. They will be discovered if you press the refresh button at the top of the test panel. They will also be rebuilt/rediscovered when you run or debug tests.

This change should be available in the next pre-release. I think a pre-release just snapped before I was able to get this change in, so it likely will be the pre-release after next (i.e. after 1.7.2).
